Initial code base

tbrent opened the pull request.

tbrent committed:
init implementation. compiles, but untested

tbrent committed:
reorder and commenting

tbrent committed:
refactor onlyWhitelist

tbrent committed:
add rebalance function signatures

tbrent committed:
add nonce

tbrent committed:
WIP still, but checkpoint anyway

tbrent committed:
factory design init checkpoint

tbrent committed:
rename functions and switch order of emits

tbrent committed:
separate accepting from enacting and add 12 hour delay

tbrent committed:
add start of extra maxIssuance logic

tbrent committed:
finish issueMax and redeemMax

tbrent committed:
dual-proposal design compiles

tbrent committed:
comments and organization

tbrent committed:
remove openzeppelin Ownable

tbrent committed:
stash for rename

tbrent committed:
Update README.md

tbrent committed:
init abi and artifacts

tbrent committed:
init reserve + eternal storage

tbrent committed:
tests passing on initial setup

tbrent committed:
top-level main contract structure

tbrent committed:
add max supply to RSV and move ownership out

tbrent committed:
README

tbrent committed:
remove abi, artifact, and soltools

tbrent committed:
move contracts out of folders, add make abi

tbrent committed:
get tests back to working

tbrent committed:
use openzeppelin IERC20 instead of in-file

fiddlemath committed:
Create an abi directory if needed, during ABI generation.

tbrent committed:
make contracts + customize OptimizeRuns per target

tbrent committed:
checkpoint: deployment succeeding with commented out code

tbrent committed:
reduce size of require messages

tbrent committed:
refactor manager auth functions for succintness

tbrent committed:
switch all modifiers to using internal functions

tbrent committed:
fix reserve tests and start manager tests

tbrent committed:
almost there! only one more function to uncomment

tbrent committed:
finalize manager under the bytecode limit

tbrent committed:
use 100 max tokens

tbrent committed:
switch proposal status to enum

tbrent committed:
fix missing status state change

tbrent committed:
add changeOwner to vault + reserve. passing manager deploy tesT

tbrent committed:
manager constructor unittest

tbrent committed:
refactor contracts into folders

tbrent committed:
move proposal events into manager

tbrent committed:
remove unnecessary max functions

fiddlemath committed:
clean target for Makefile

fiddlemath committed:
Change "abi" generation to output solc combined-json files into evm/

This version continues also to emit the JSON format expected by sol-coverage; those
now get made in sol-coverage-evm/

fiddlemath committed:
trivial conflict resolution

fiddlemath committed:
Added current draft of design docs, with use cases and security properties.. So drafty!

tbrent committed:
add requireTxStrongly/Weakly

fiddlemath committed:
Added detailed notes on deployment.

fiddlemath commented:
Why use Solidity 0.5.8 instead of the most recent version?

fiddlemath reviewed (commented):
Why use Solidity 0.5.8 instead of the most recent version?

tbrent commented:
No good reason.

fiddlemath committed:
Set gitattributes so that Solidity files are syntax-highlighted on github.

tbrent committed:
add basket swap

tbrent committed:
remove changeOwner from Reserve

tbrent committed:
add auth protection to executeProposal

tbrent committed:
fix basket swap ordering bug in executeProposal

tbrent committed:
test for proposeNewBasket

tbrent committed:
fix basket bugs and simplify

tbrent committed:
safemath for proposal math

tbrent committed:
make manager fully collateralized check public

tbrent committed:
fix Ownable error message

tbrent committed:
explain modifier approach in comments

tbrent committed:
add missing require

tbrent committed:
make delay an auth parameter

tbrent committed:
test updates for easier testing of proposal refactor

tbrent committed:
make auth control uniform

tbrent committed:
add BasketChanged event

tbrent committed:
add ManagerTransferred event to vault constructor

fiddlemath committed:
Merging.

tbrent committed:
change vault events

tbrent committed:
add vault tests

tbrent committed:
finish vault tests

tbrent committed:
fix Ownable bug with accepting ownership

tbrent committed:
add completed ownable tests

fiddlemath committed:
Local copy of pull review checklist

fiddlemath committed:
A snapshot

tbrent committed:
remove changeName from RSV

tbrent committed:
remove freezing/wiping RSV functionality

tbrent committed:
Merge branch 'working' of github.com:reserve-protocol/rsv-beta into working

tbrent committed:
remove whitelisting + add manager tests

tbrent committed:
fix typos in basket

tbrent committed:
complete refactor; compiles

tbrent committed:
fix new weight proposal bug

fiddlemath committed:
Draft of manager with new basket

fiddlemath committed:
Somehwat cleaner build messages from generate.go

fiddlemath committed:
Compiling version of proposal redesign

fiddlemath committed:
Merge branch 'refactor-proposal-and-basket' of github.com:reserve-protocol/rsv-beta into refactor-proposal-and-basket

fiddlemath committed:
Post-merge fixes to reach compilation

fiddlemath committed:
Merge branch 'working' into refactor-proposal-and-basket

fiddlemath committed:
Merge pull request #6 from reserve-protocol/refactor-proposal-and-basket

Refactor proposal and basket

fiddlemath committed:
Just tweaking some comments

fiddlemath committed:
Cleaning up most remaining TODOs.

The only remaining TODO in the contract comments is to think very clearly about rounding error,
which is a serious task rather than a quick little fix.

fiddlemath committed:
Nitpicky comment edit

tbrent committed:
remove redundant type cast

tbrent committed:
remove outdated TODO comment

tbrent committed:
add comments for proposal contracts

tbrent committed:
remove last instance of unnecessary underscore names

tbrent committed:
restore tests to where they were before

tbrent committed:
fix bug with manager not being able to pass its first proposal

tbrent committed:
combine executeProposal events

tbrent committed:
manager first proposal test pass

tbrent committed:
only we can execute proposals

tbrent committed:
small comment update

tbrent committed:
comment tweaks

tbrent committed:
undo accidentally checked in merge characters

fiddlemath committed:
Handle rounding in RSV weighting.

tbrent committed:
basket tests

tbrent committed:
Merge branch 'working' of github.com:reserve-protocol/rsv-beta into working

fiddlemath committed:
Draft of handling rounding in SwapProposals

fiddlemath committed:
Safemath for mod; remove vault reference from proposal code

fiddlemath committed:
Fixed up rounding a bit, and reordered Manager

tbrent committed:
progress on manager test and init proposal test

fiddlemath committed:
Changed isFullyCollateralized to need no rounding and be easier to reason with.

fiddlemath committed:
Changed weights to be 36-digit fixed point instead of 18-digit fixed point.

tbrent committed:
fix small comment typo

tbrent committed:
add BasketCreated event to proposals

fiddlemath committed:
Improve comments to present correct units and dimensional analysis.

tbrent committed:
rename proposal event

tbrent committed:
check-in confusing failing test

fiddlemath committed:
Do not override proposer in SwapProposal

fiddlemath committed:
Remove comment indicating spurious uncertainty!

fiddlemath committed:
Clean up Basket header comment

fiddlemath committed:
Add Basket::constructor comment

fiddlemath committed:
Subtract 1 from token amount in SwapProposal, to avoid overspending upon execution.

tbrent committed:
add proposal comment

tbrent committed:
proposal tests

fiddlemath committed:
Make Manager.issue and Manager.redeeem external instead of public.

tbrent committed:
make IRSV same as IERC20

fiddlemath committed:
Cleanup Vault by removing unused stuff, and inlining a modifier used only once.

fiddlemath committed:
Add an explicit uint type conversion. Pants and suspenders!

fiddlemath committed:
RSV contract review improvements:

- Bring comments in line with rest of contracts
- Drop frozenTime
- Reorder some things to be easier to read
- Change TXFee handling slightly

fiddlemath committed:
slither triage file.

Handled correctly, this might make it much easier to use slither to double-check needed properties.
On the other hand, maybe you want to make your own choices about what warnings to attend.

fiddlemath committed:
Remove stale README and Makefile

tbrent committed:
Update contracts/rsv/Reserve.sol

Co-Authored-By: Matt Elder <fiddlemath@users.noreply.github.com>

fiddlemath committed:
Do not allow owner to call acceptOwnership

fiddlemath committed:
tx fee helper's calculateFee should take from and to addresses.

fiddlemath committed:
Use solc 0.5.7 throughout this project.

Why 0.5.7? It's the oldest version that:
- supports all the features our code actually uses (the extcodehash op)
- was not shortly followed by bugfixes (like 0.5.5 and 0.5.6 were)

tbrent committed:
tests!

The tests don't yet comprehensively cover Manager, and they also don't test the math precisely.

fiddlemath committed:
Merge branch 'elder-friday-night-pr' into working

fiddlemath committed:
Merge branch 'working' of github.com:reserve-protocol/rsv-beta into working
